source/fontloader.ts

```diff
@@ -0,0 +1,289 @@
+        const pairs = this.readKeyValuePairs(stream, ['first', 'second', 'amount']);
+
+        const first: number = parseInt(pairs.get('first')!, 10);
+        logIf(first < 0.0, LogLevel.Error,
+            `expected kerning's first to be greater than 0, given ${first}`);
+
+        const second: number = parseInt(pairs.get('second')!, 10);
+        logIf(second < 0.0, LogLevel.Error,
+            `expected kerning's first to be greater than 0, given ${second}`);
+
+        const kerning: number = parseFloat(pairs.get('amount')!);
+
+        fontFace.setKerning(first, second, kerning);
+    }
+
+    /**
+     * Parses to find key-value pairs for given mandatory keys.
+     * @param stream The stream from which the pairs should be read.
+     * @param mandatoryKeys The found pairs are only valid if the mandatory keys are found.
+     * @returns key-value pairs, or an empty map if not all mandatory keys are found.
+     */
+    protected static readKeyValuePairs(stream: Array<string>, mandatoryKeys: Array<string>): StringPairs {
+
+        let key: string;
+        let value: string;
+
+        const pairs: StringPairs = new Map<string, string>();
+
+        for (const s of stream) {
+            const pair = s.split('=');
+            key = pair[0];
+            value = pair[1];
+            pairs.set(key, value);
+        }
+
+        /* check if all required keys are provided */
+        let valid = true;
+        mandatoryKeys.forEach((key) => valid = valid && pairs.has(key));
+
+        if (!valid) {
+            log(LogLevel.Warning, `Not all required keys are provided! Mandatory keys: ${mandatoryKeys}`);
+            /* typescript does not allow writing `new StringPairs()` */
+            return new Map<string, string>();
+        } else {
+            return pairs;
+        }
+    }
+
+    /**
+     * Asynchronously loads a fnt-file and a png-file of the same name, to create a font face from them.
+     * @param context The WebGL rendering context.
+     * @param filename The path to the fnt-file.
+     * @param headless Boolean for headless mode.
+     * @param onImageLoad Callback is called when the glyph atlas is loaded.
+     */
+    static async load(context: Context, filename: string, headless: boolean): Promise<FontFace> {
+        const fontFace = new FontFace(context);
+
+        try {
+            const text = await fetchAsync(filename, '', (text) => text);
+
+            /* promise fulfilled */
+            const lines = text.split('\n');
+
+            const promises = [];
+            for (const l of lines) {
+                let line = l.split(' ');
+                const identifier = line[0];
+                line = line.slice(1);
+
+                /* tslint:disable-next-line:switch-default */
+                switch (identifier) {
+                    case 'info': {
+                        this.processInfo(line, fontFace);
+                        break;
+                    }
```

Comment:
you mean the curly brackets? I put them because the style requires the curly brackets even for if-oneliners. I can remove them, no problem.

Reply:
hmm .. `auxiliaries.ts` lines 22ff, `color.ts` lines 327ff, 521ff, `gl2facade.ts` lines 164ff, etc... can you please open one of those and check if your local linter complains - if so, there might be an issue with the linter setup - or run `npm run-script lint` and check if it occurs..

Reply:
no, that's fine. What I was trying to say is that I didn't look for other occurences of switch-cases, but immediately wrote the curly brackets because in similar cases (e.g., if-else), the linter requires curly brackets. 
I removed them now for the switch cases.